esp-phy/src/common_adapter.rs

```diff
@@ -0,0 +1,77 @@
+use esp_hal::ram;
+
+/// **************************************************************************
+/// Name: phy_enter_critical
+///
+/// Description:
+///   Enter critical state
+///
+/// Input Parameters:
+///   None
+///
+/// Returned Value:
+///   CPU PS value
+///
+/// *************************************************************************
+#[ram]
+#[unsafe(no_mangle)]
+unsafe extern "C" fn __esp_phy_enter_critical() -> u32 {
+    trace!("phy_enter_critical");
+
+    unsafe { core::mem::transmute(critical_section::acquire()) }
```

Comment:
That's not what the old code looks like, can you re-replace critical_section with esp-sync locks? I really don't want to see the critical-section dependency here.